syntax/printer.go

```diff
@@ -1081,6 +1081,13 @@ func (p *Printer) command(cmd Command, redirs []*Redirect) (startRedirs int) {
 		p.WriteByte('(')
 		if len(x.Stmts) > 0 && startsWithLparen(x.Stmts[0]) {
 			p.wantSpace = spaceRequired
+			if nested, ok := x.Stmts[0].Cmd.(*Subshell); ok {
```

Comment:
I would do this as part of startsWithLparen, which already looks into the type of the first command. Moreover, it multiple node types, which you will want to do as well (add tests too, please).